src/renderer/stores/sequencer.ts

```diff
@@ -2,36 +2,70 @@ import { v4 as uuidv4 } from "uuid";
 import { create } from "zustand";
 import { immer } from "zustand/middleware/immer";
 import {
+  BackendGlobalState,
+  CycleConfig,
   MsgState,
+  StatusType,
   TestRootNode,
+  TestSequenceContainer,
   TestSequenceElement,
+  UploadInfo,
 } from "@/renderer/types/test-sequencer";
 import { TestSequencerProject } from "@/renderer/types/test-sequencer";
+import { testSequenceRunRequest } from "@/renderer/routes/test_sequencer_panel/models/models";
+import { toast } from "sonner";
+import { createTestSequenceTree } from "@/renderer//hooks/useTestSequencerState";
+import { SendJsonMessage } from "react-use-websocket/dist/lib/types";
 
 type State = {
-  curRun: string[];
   websocketId: string;
   elements: TestSequenceElement[];
   isLocked: boolean;
   isLoading: boolean;
-  backendState: MsgState;
+  backendGlobalState: BackendGlobalState;
+  cycleConfig: CycleConfig;
+  cycleRuns: TestSequenceContainer[][];
+  playPauseState: MsgState;
   testSequenceUnsaved: boolean;
-  testSequenceTree: TestRootNode;
-  testSequencerProject: TestSequencerProject | null;
+  testSequenceStepTree: TestRootNode;
+  testSequenceDisplayed: TestSequencerProject | null;
+  sequences: TestSequenceContainer[];
+  uploadAfterRun: boolean;
+  uploadInfo: UploadInfo;
 };
 
 type Actions = {
-  markTestAsDone: (val: string) => void;
-  addTestToRunning: (val: string) => void;
-
   setWebsocketId: (val: string) => void;
   setElements: (val: TestSequenceElement[]) => void;
   setIsLocked: (val: boolean) => void;
   setIsLoading: (val: boolean) => void;
+  setBackendGlobalState: (val: BackendGlobalState) => void;
   setBackendState: (val: MsgState) => void;
   setTestSequenceUnsaved: (val: boolean) => void;
   setTestSequenceTree: (val: TestRootNode) => void;
-  setTestSequencerProject: (val: TestSequencerProject | null) => void;
+  setUploadAfterRun: (val: boolean) => void;
+  setSerialNumber: (val: string) => void;
+  setIntegrity: (val: boolean) => void;
+  setStationId: (val: string) => void;
+  setIsUploaded: (val: boolean) => void;
+  // Cycles
+  setCycleCount: (val: number) => void;
+  setInfinite: (val: boolean) => void;
+  saveCycle: () => void;
+  diplayPreviousCycle: () => void;
+  displayNextCycle: () => void;
+  clearPreviousCycles: () => void;
+  // Sequences
+  setSequences: (val: TestSequenceContainer[]) => void;
+  addNewSequence: (
+    val: TestSequencerProject,
+    elements: TestSequenceElement[],
+  ) => void;
+  removeSequence: (name: string) => void;
+  displaySequence: (name: string) => void;
+  runNextRunnableSequence: (sender: SendJsonMessage) => void;
+  runRunnableSequencesFromCurrentOne: (sender: SendJsonMessage) => void;
+  updateSequenceStatus: (val: StatusType) => void;
 };
 
 export const useSequencerStore = create<State & Actions>()(
```

Comment:
a lot of this store feels like misuse of zustand
if you want to access do this outside of the `set` callback then you can add `get` (`immer((set, get) => ...`) which is just a function that returns the current state.